.github/workflows/ci.yaml

```diff
@@ -1,8 +1,10 @@
 name: CI
 
 'on':
-  - push
-  - pull_request
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
Do we want this to run on pull requests?

Reply:
This workflow isn't deploying to GitHub pages; it's just running a build and the linters. So this is the workflow we use to make sure that PRs are in good order. The `deploy.yaml` worklow is what also does the deployment to `gh-pages`.